noxfile.py

```diff
@@ -19,7 +20,19 @@ def tests_impl(
     extras: str = "socks,secure,brotli",
     byte_string_comparisons: bool = True,
 ) -> None:
+    is_github_actions = "GITHUB_ACTIONS" in os.environ
+
     # Install deps and the package itself.
+    # Compile cryptography with the system OpenSSL and libffi libraries
+    # on Linux CPython runs of GitHub Actions instead of downloading its
+    # wheel.
+    is_cpython_on_linux = (
```

Comment:
I like the direction this is heading, I wonder if there's a way to detect if we're running against a Python with OpenSSL 3.0 (maybe `python -c 'import ssl; print(ssl.OPENSSL_VERSION[0])'` and capturing the output with `nox.Session.run(..., stdout=...)`? Then if we detect OpenSSL 3.0 we run `session.install("--no-binary=:all:", "cryptography")` to install from source as you're already doing now.